examples/js/archive.js

```diff
@@ -1,125 +1,124 @@
-        getUrl = `https://${input.value}`;
-        showImages(getUrl);   
-      }
-      else if(input.value.includes('http://')) {
-        getUrl = input.value.replace('http:', 'https:');
-        input.value = getUrl;
-        console.log('input', input.value);
-        showImages(getUrl);   
-      }
-      else
-      {
-        getUrl = input.value;
-        showImages(getUrl)
-      }
-    }
-    
-    function showImages(getUrl) {
-      const url = getUrl.replace('details', 'metadata');
-      let fetchedFrom
-      axios.get(url)
+let map;
+const welcomeModal = document.getElementById('welcomeModal');
+const sidebar = document.getElementById('offcanvasRight');
+const form = document.getElementById('form');
+const input = document.getElementById('input');
+const responseText = document.getElementById('response');
+const imageContainer = document.getElementById('imgContainer');
+const mapToggle = document.getElementById('mapToggle');
+
+const setupMap = () => {
+  map = L.map('map').setView([51.505, -0.09], 13);
+
+  map.attributionControl.setPosition('bottomleft');
+
+  map.addGoogleMutant();
+  map.whenReady(() => {
+    new bootstrap.Modal(welcomeModal).show();
+  });
+};
+
+const setupCollection = () => {
+  map.imgGroup = L.distortableCollection().addTo(map);
+};
+
+setupMap();
+L.Control.geocoder().addTo(map);
+setupCollection();
+
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+  extractKey();
+});
+
+welcomeModal.addEventListener('hidden.bs.modal', (event) => {
+  new bootstrap.Offcanvas(sidebar).show();
+});
+
+function extractKey() {
+  let getUrl;
+  if (!input.value.includes('archive.org/details/') && !input.value.includes('https://'))
+  {
+    getUrl = `https://archive.org/details/${input.value}/`;
+    showImages(getUrl);
+  }
+  else if (!input.value.includes('https://') && !input.value.includes('http://') && input.value.includes('archive.org/details/')) {
+    getUrl = `https://${input.value}`;
+    showImages(getUrl);
+  }
+  else if (input.value.includes('http://')) {
+    getUrl = input.value.replace('http:', 'https:');
+    input.value = getUrl;
+    console.log('input', input.value);
+    showImages(getUrl);
+  }
+  else
+  {
+    getUrl = input.value;
+    showImages(getUrl);
+  }
+}
+
+function showImages(getUrl) {
+  const url = getUrl.replace('details', 'metadata');
+  let fetchedFrom;
+  axios.get(url)
       .then((response) => {
         if (response.data.files && response.data.files.length != 0) {
           let imageCount = 0;
-          response.data.files.forEach(file => {
-            if (file.format === 'PNG' || file.format === 'JPEG'){
-              let imageRow = document.createElement('div');
-              let image = new Image(150, 150);
-              let placeButton = document.createElement('a');
+          response.data.files.forEach((file) => {
```

Comment:
This looks great. I'm going to merge it -- maybe a next step could be to break out lines 70-96 in a separate function called, what do you think... `renderImages`? or would it be better to have separate ones like `renderPng()` and `renderJpeg()`? Yeah, maybe that sounds better?

Reply:
Oh, that sounds a lot better @jywarren 
I'd add that next. Thanks for the review

Reply:
I was just going through this again, i think `renderImages`  would be better as there is no distinction when we render a .png or .jpeg file as noticed here:

https://github.com/7malikk/Leaflet.DistortableImage/blob/b396b5a0023183605fa4354626c886638659bf34/examples/js/archive.js#L69-L92

There is no different render method depending on file type.

What do you think @jywarren?